sources/packages/api/src/route-controllers/user/user.controller.ts

```diff
@@ -35,18 +34,22 @@ export class UserController extends BaseController {
     const account = await this.bceidService.getAccountDetails(
       userToken.idp_user_name,
     );
-    return {
-      user: {
-        guid: account.user.guid,
-        displayName: account.user.displayName,
-        firstname: account.user.firstname,
-        surname: account.user.surname,
-        email: account.user.email,
-      },
-      institution: {
-        guid: account.institution.guid,
-        legalName: account.institution.legalName,
-      },
-    };
-  }
+    if (account == null) {
+      return {} as BCeIDDetailsDto;
```

Comment:
Can you elaborate more about why is it better to return an empty object instead of null?
It could also return a 404.

Reply:
Returning 404 from API forces me to do this in the web/src/services/http/UserApi.ts and i do not like what i am having to do in catch block. Check this issue https://github.com/axios/axios/issues/960

public async bceidAccount(headers?: any): Promise<BCeIDDetailsDto | null> {
    try {
      const response = await this.apiClient.get(
        "users/bceid-account",
        headers || this.addAuthHeader(),
      );
      return response.data as BCeIDDetailsDto;
    } catch (error) {
      const errorObject = JSON.parse(JSON.stringify(error));
      if (errorObject.message.includes("404")) {
        return null;
      } else {
        this.handleRequestError(error);
        throw error;
      }
    }
  }
  
  Returning a null

Reply:
@2011sandeepraj , There is **error.response** object which has **status** property which gives you exact https status code. Try to explore **error.response** object.

Even you can directly check **error.message**, JSON parsing  is pure unnecessary overhead.